File: foxogram-common/src/main/java/su/foxogram/dto/gateway/response/HelloDTO.java
package su.foxogram.dto.gateway.response;

import lombok.Getter;
import lombok.Setter;
import su.foxogram.constant.GatewayConstant;

import java.util.Map;

@Getter
@Setter
public class HelloDTO {

	private int op;

	private Map<String, Integer> d;

	public HelloDTO() {
		this.op = GatewayConstant.Opcode.HELLO.ordinal();
		this.d = Map.of("heartbeat_interval", GatewayConstant.HEARTBEAT_INTERVAL);
	}
}

File: src/main/java/su/foxogram/dto/gateway/response/HelloDTO.kt
package su.foxogram.dto.gateway.response

import lombok.Getter
import lombok.Setter
import su.foxogram.constant.GatewayConstant
import java.util.Map

@Getter
@Setter
data class HelloDTO(
    val op: Int = GatewayConstant.Opcode.HELLO.ordinal,

    val d: MutableMap<String, Int>? = Map.of<String, Int>("heartbeat_interval", GatewayConstant.HEARTBEAT_INTERVAL)
)
